source/linux/Foundational_Components/Tools/Flash_via_Fastboot.rst

```diff
@@ -0,0 +1,126 @@
+
+The Snagfactory tool has two main operations:
+
+* Snagrecover: Boots the board (recovery) using USB DFU.
+* Snagflash: Flashes binaries to the on-board memory using the Fastboot protocol.
+
+This tool supports flashing multiple boards simultaneously, enhancing efficiency in producion
+enviroment.
+
+********
+Get Tool
+********
+
+* Snagfactory tool is hosted here `Snagfactory <https://github.com/bootlin/snagboot>`__.
+* More info about installation can be found in `Snagfactory Readme <https://github.com/bootlin/snagboot/blob/main/README.md>`__.
+* Snagfactory also is available on pip.
+
+.. code-block:: python
+
+    python3 -m pip install --user snagboot
+
+*****************************************
+Building bootloader binaries for Recovery
+*****************************************
+
+For Snagrecover, bootloader images must support DFU boot and fastboot download.
+In addition to USB DFU fragment config (which enables DFU boot) for the u-boot
+build, an additional fragment config *am6x_a53_snagfactory.config* needs to be
+used, which enables fastboot support in U-Boot and other required configs for
+snagfactory.
+
+To build bootloader images for recovery using SDK, following change is needed
+in :file:`Rules.make` file present in the top level of Linux SDK Installer.
+
+.. ifconfig:: CONFIG_part_variant in ('AM62X')
+
+    .. code-block:: make
+
+        UBOOT_MACHINE_R5=am62x_evm_r5_defconfig am62x_r5_usbdfu.config
+
+        UBOOT_MACHINE_A53=am62x_evm_r5_defconfig am62x_a53_usbdfu.config am6x_a53_snagfactory.config
+
+        # For AM62X LP
+
+        UBOOT_MACHINE_R5=am62x_lpsk_r5_defconfig am62x_r5_usbdfu.config
+
+        UBOOT_MACHINE_A53=am62x_lpsk_a53_defconfig am62x_a53_usbdfu.config am6x_a53_snagfactory.config
+
+        # For AM62X SIP
+
+        UBOOT_MACHINE_R5=am62xsip_evm_r5_defconfig am62x_r5_usbdfu.config
+
+        UBOOT_MACHINE_A53=am62xsip_evm_a53_defconfig am62x_a53_usbdfu.config am6x_a53_snagfactory.config
+
+.. ifconfig:: CONFIG_part_variant in ('AM64X')
+
+    .. code-block:: make
+
+        UBOOT_MACHINE_R5=am64x_evm_r5_defconfig
+
+        UBOOT_MACHINE_A53=am64x_evm_a53_defconfig am6x_a53_snagfactory.config
+
+.. ifconfig:: CONFIG_part_variant in ('AM62AX')
+
+    .. code-block:: make
+
+        UBOOT_MACHINE_R5=am62ax_evm_r5_defconfig am62x_r5_usbdfu.config
+
+        UBOOT_MACHINE_A53=am62ax_evm_a53_defconfig am62x_a53_usbdfu.config am6x_a53_snagfactory.config
+
+.. ifconfig:: CONFIG_part_variant in ('AM62PX')
+
+    .. code-block:: make
+
+        UBOOT_MACHINE_R5=am62px_evm_r5_defconfig am62x_r5_usbdfu.config
+
+        BOOT_MACHINE_A53=am62px_evm_a53_defconfig am62x_a53_usbdfu.config am6x_a53_snagfactory.config
+
+Generate the bootloader images using top-level makefile by running following
+commands on the terminal from the top-level of the Linux SDK installer.
+
+.. code-block:: console
+
+    $ make u-boot_clean
+    $ make u-boot
```

Comment:
Better to add `make u-boot_stage` as well and ask users to get the binaries from `/board-support/built-images`